lib/vanagon/engine/scheduled.rb

```diff
@@ -0,0 +1,27 @@
+require 'vanagon/engine/base'
+require 'json'
+
+class Vanagon
+  class Engine
+    # Class to use when building on a shared resource device that is managed
+    # by an external scheduler.
+    class Scheduled < Base
```

Comment:
What about using the `Scheduled` engine for both `pooler` and `hardware`? The [`build_host_name` method below](https://github.com/puppetlabs/vanagon/pull/417/files#diff-b48dd1920d5ccbd8b89b388eda6f547eR22) could return the `template_name` for pooler platforms and [something else](https://github.com/puppetlabs/vanagon/pull/417/files#r85019195) for hardware platforms.

If we only use `Scheduled` for hardware platforms, then `engine` can either be `pooler` (which is really the default/preferred allocator) or `scheduled` (which says to use the scheduler, but omits the allocator). IOW, engine means different things in different contexts, which gets confusing.